### PR TITLE
Add preferredOptions

### PR DESCRIPTION
**Pull request by bsudekum**
@ericdeveloper and I were talking about an easier way for developers to get started. RouteOptions is particularly a tricky configuration to get right since it requires inside knowledge of the directions API to get it right.

This PR moves the options out into an extension so the developer does not need to worry about these options. 

/cc @1ec5 @frederoni @ericrwolfe

### CODE REVIEW

File: Examples/Swift/ViewController.swift [hunk 1]
```diff
@@ -270,12 +266,8 @@ extension ViewController: WaypointConfirmationViewControllerDelegate {
...
             nextDestination,
         ])
-        options.includesSteps = true
-        options.routeShapeResolution = .full
-        options.profileIdentifier = navigationViewController.route.routeOptions.profileIdentifier
-        
-        // Adding the optional attribute `.congestionLevel` ensures the route line will show the congestion along the route line
-        options.attributeOptions = [.congestionLevel]
+            // Dont forget to apply `preferredOptions`. This adds the necessary options needed to make a navigation experience 💯.
+            .preferredOptions
```
Comment (ericrwolfe) [inline Examples/Swift/ViewController.swift]: Should this instead be a custom initializer? Maybe something like `RouteOptions(forNavigationWithCoordinates:)`? Feels weird to have `.preferredOptions` modify the existing options. Would be confusing if it overrode something the developer wanted to set explicitly.

Comment (frederoni) [inline Examples/Swift/ViewController.swift]: I agree about the overriding. However, [coordinates] won't bridge to Obj-C, how about a class function like `preferredOptions(waypoints:profileIdentifier:)`?

**Comment by bsudekum** [comment]
Updated to use init instead.

File: MapboxCoreNavigation/RouteOptions.swift [hunk 2]
```diff
@@ -9,4 +9,31 @@ extension RouteOptions {
...
         }
     }
+    
+    /**
+     Returns an optimal `RouteOptions` for navigation.
+     */
+    public convenience init(forNavigationWithWaypoints waypoints: [Waypoint], profileIdentifier: MBDirectionsProfileIdentifier? = .automobileAvoidingTraffic) {
+        self.init(waypoints: waypoints, profileIdentifier: profileIdentifier)
+        
+        includesSteps = true
```
Comment (bsudekum) [inline MapboxCoreNavigation/RouteOptions.swift]: @frederoni is there a way we can store these options once in this file?

Comment (1ec5) [inline MapboxCoreNavigation/RouteOptions.swift]: Why not subclass RouteOptions as NavigationOptions? Then NavigationOptions could store its own properties. This approach is intended to be supported by MapboxDirections.swift, and it has the benefit of being more discoverable and more easily documentable.

**Pull request merged**